runtime/runtime.go

```diff
@@ -77,19 +80,39 @@ type Runtime interface {
 
 	// GetEnv returns map of environment variables read from run.yaml.
 	GetEnv() map[string]string
+
+	// ForceEnv updates runtime's environment with given values.
+	// List of updated keys is returned
+	ForceEnv(env map[string]string) []string
 }
 
 // CommonRuntime fields are those common to all runtimes.
 // This fields are set for each named-configuration separately, nothing
 // is shared.
 type CommonRuntime struct {
-	Env map[string]string `yaml:"env"`
+	Env  map[string]string `yaml:"env"`
+	Base string            `yaml:"base"`
 }
 
 func (r CommonRuntime) GetEnv() map[string]string {
 	return r.Env
 }
 
+// ForceEnv updates runtime's own environment with given values.
+// Only existing keys are updated and then a list of those is returned.
+// Purpose of this function is to be able to modify environment (that was
+// parsed from meta/run.yaml) just before bootcmd is generated.
+func (r CommonRuntime) ForceEnv(env map[string]string) []string {
+	updated := make([]string, 15)
```

Comment:
Again `15`? In this case it is even worse, because you actually create an array of length 15. If you just need capacity, you should use `make([]string, 0, 15)` (contrary to the map, the second parameter here is length).